ssg/build_yaml.py

```diff
@@ -111,7 +112,17 @@ def __init__(self, id_):
         self.unselected = []
         self.variables = dict()
         self.refine_rules = defaultdict(list)
-        self.policies = {}
+
+    def read_yaml_contents(self, yaml_contents):
```

Comment:
This has been extracted from `from_yaml`, and it allows extension of yaml processing in derived classes.